### ISSUE TITLE
Initial provisioning is currently broken

### ISSUE THREAD

**Issue by rmccue** [humanmade/platform_chassis_extension#10]
There's some sort of weird provisioning issue currently. On the first provision, it fails after adding the PPAs with:

```
==> default: Notice: /Stage[main]/Chassis::Php/Package[software-properties-common]/ensure: ensure changed '0.96.20.7' to '0.96.20.8'                                                                      
==> default: Notice: /Stage[main]/Chassis::Php/Apt::Ppa[ppa:ondrej/php]/Exec[add-apt-repository-ppa:ondrej/php]/returns: executed successfully                                                            
==> default: Notice: /Stage[main]/Chassis::Php/Apt::Pin[php7.2-fpm]/Apt::Setting[pref-php7.2-fpm]/File[/etc/apt/preferences.d/php7.2-fpm.pref]/ensure: created                                            
==> default: Notice: /Stage[main]/Chassis::Php/Apt::Pin[php7.2-gd]/Apt::Setting[pref-php7.2-gd]/File[/etc/apt/preferences.d/php7.2-gd.pref]/ensure: created                                               
==> default: Notice: /Stage[main]/V8js::Extension/Apt::Ppa[ppa:pinepain/libv8]/Exec[add-apt-repository-ppa:pinepain/libv8]/returns: executed successfully                                                 
==> default: Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install libv8-6.6-dev' returned 100: Reading package lists...                                                 
==> default: Building dependency tree...                                                                                                                                                                  
==> default: Reading state information...                                                                                                                                                                 
==> default: E: Unable to locate package libv8-6.6-dev
==> default: E: Couldn't find any package by glob 'libv8-6.6-dev'                                                                                                                                         
==> default: E: Couldn't find any package by regex 'libv8-6.6-dev'
==> default: Error: /Stage[main]/V8js::Extension/Package[libv8-6.6-dev]/ensure: change from purged to present failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install libv8-6.6-dev' returned 100: Reading package lists...
```

Other packages also fail to install (abbreviated):
```
==> default: Error: /Stage[main]/V8js::Extension/Package[libv8-6.6]/ensure: change from purged to present failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install libv8-6.6
==> default: Error: /Stage[main]/V8js::Extension/Package[php7.2-dev]/ensure: change from purged to present failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php7.2-dev' returned 100: Reading package lists...
==> default: Error: /Stage[main]/Chassis::Php/Package[php7.2-zip]/ensure: change from purged to latest failed: Could not update: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold --force-yes install php7.2-zip' returned 100: Reading package lists...
```

etc.

However, a second `vagrant provision` works, indicating there's some sort of missing dependency in here somewhere. Provisioning was working last week, so I suspect this may be from the latest updates? cc @roborourke

**Comment by roborourke** [comment]
@rmccue I saw this too but it points to v8js again, running a fresh install without it to double check

**Comment by rmccue** [comment]
This is somehow caused by the Docker module in chassis-aws-analytics, which is moving an Apt::Update refresh much later.

Here's a working provision:
```
[...]
==> default: Notice: /Stage[main]/Chassis::Php/Apt::Pin[php7.2-fpm]/Apt::Setting[pref-php7.2-fpm]/File[/etc/apt/preferences.d/php7.2-fpm.pref]/ensure: created                                            
==> default: Notice: /Stage[main]/Chassis::Php/Apt::Pin[php7.2-gd]/Apt::Setting[pref-php7.2-gd]/File[/etc/apt/preferences.d/php7.2-gd.pref]/ensure: created                                               
==> default: Notice: /Stage[main]/V8js::Extension/Apt::Ppa[ppa:pinepain/libv8]/Exec[add-apt-repository-ppa:pinepain/libv8]/returns: executed successfully                                                 
==> default: Notice: /Stage[main]/Apt::Update/Exec[apt_update]: Triggered 'refresh' from 1 events
```

And a broken one:
```
[...]
==> default: Notice: /Stage[main]/Chassis::Php/Apt::Pin[php7.2-fpm]/Apt::Setting[pref-php7.2-fpm]/File[/etc/apt/preferences.d/php7.2-fpm.pref]/ensure: created                                            
==> default: Notice: /Stage[main]/Chassis::Php/Apt::Pin[php7.2-gd]/Apt::Setting[pref-php7.2-gd]/File[/etc/apt/preferences.d/php7.2-gd.pref]/ensure: created                                               
==> default: Notice: /Stage[main]/V8js::Extension/Apt::Ppa[ppa:pinepain/libv8]/Exec[add-apt-repository-ppa:pinepain/libv8]/returns: executed successfully                                                 
==> default: Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install libv8-6.6-dev' returned 100: Reading package lists...
```

In the broken one, we can see Apt::Update triggered later:
```
==> default: Notice: /Stage[main]/Docker::Repos/Apt::Source[docker]/Apt::Key[Add key: 58118E89F3A912897C070ADBF76221572C52609D from Apt::Source docker]/Apt_key[Add key: 58118E89F3A912897C070ADBF76221572C
52609D from Apt::Source docker]/ensure: created                                                                                                                                                            
==> default: Notice: /Stage[main]/Docker::Repos/Apt::Source[docker]/Apt::Setting[list-docker]/File[/etc/apt/sources.list.d/docker.list]/ensure: created                                                    
==> default: Notice: /Stage[main]/Apt::Update/Exec[apt_update]: Triggered 'refresh' from 1 events
```

**Comment by rmccue** [comment]
I attempted to upgrade the apt module in Chassis itself, but that didn't fix this, so I have no idea why this is happening.

**Comment by roborourke** [comment]
For the sake of time I'll switch it to node now - PR shortly.

**Comment by roborourke** [comment]
Fixed in humanmade/chassis-aws-analytics#3
